### PR TITLE
Feature/drawer

### PR DESCRIPTION
**Pull request by davewish**

### CODE REVIEW

File: src/contexts/SettingProvider.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import PropTypes from "prop-types";
```
Comment (kuldep7) [inline src/contexts/SettingProvider.jsx:1]: Please check AuthContext file to follow,
basically no nee to separate context and provider

File: src/hooks/useSetting.js [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+  const fetchSetting = useCallback(async () => {
+    if (!isMounted.current) {
+      isMounted.current = true;
+      return;
+    }
+    dispatch({ type: actionTypes.FETCH_START });
+
+    try {
+      const response = await axiosPrivate.get(url);
+      dispatch({ type: actionTypes.FETCH_SUCCESS, payload: response?.data });
```
Comment (kuldep7) [inline src/hooks/useSetting.js:51]: for apis please check useGridData changes by me
we need to use "/api" now as base

File: src/contexts/SettingProvider.jsx [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+import SettingContext from "./settingContext";
+
+const url = "28912ddd5410287eb4c0";
+const SettingProvider = ({ children }) => {
+  const { state, dispatch } = useSetting(url);
+
+  return state.setting.menu.length > 0 ? (
+    <SettingContext.Provider value={{ state, dispatch }}>{children}</SettingContext.Provider>
+  ) : null;
```
Comment (kuldep7) [inline src/contexts/SettingProvider.jsx:13]: Please check AuthContext file to follow,
basically no need to separate context and provider
and in context folder please use PascalCase naming and only have XyzContext namings

**Review by kuldep7 — CHANGES_REQUESTED** [review]

File: src/contexts/LayoutProvider.jsx [hunk 4]
```diff
@@ -1,8 +1,8 @@
-import { useReducer } from "react";
+import { useReducer, createContext } from "react";
 
 import PropTypes from "prop-types";
 
-import LayoutContext from "./layoutContext";
+export const LayoutContext = createContext({ state: { open: false }, dispatch: () => {} });
```
Comment (kuldep7) [inline src/contexts/LayoutProvider.jsx:5]: no need to export context as such like we can create a hook for this context , please refer AuthProvider file

**Review by kuldep7 — CHANGES_REQUESTED** [review]

File: src/constants/endpoints.json
Comment (misghna) [inline src/constants/endpoints.json:1]: I would rename 
setting to globalSettings
GridData to contentData

Comment (davewish) [reply src/constants/endpoints.json:1]: okay

**Pull request merged**